client/tests/integration/add_account.rs

```diff
@@ -13,14 +13,14 @@ fn client_add_account_with_name_length_more_than_limit_should_not_commit_transac
     let pipeline_time = super::Configuration::pipeline_time();
 
     let normal_account_id: AccountId = "bob@wonderland".parse().expect("Valid");
-    let create_account = RegisterExpr::new(Account::new(normal_account_id.clone(), []));
+    let create_account = Register::account(Account::new(normal_account_id.clone(), []));
```

Comment:
What I want to see is this:
```suggestion
    let create_account = Register::new(Account::new(normal_account_id.clone(), []));
```

continuing [this](https://github.com/hyperledger/iroha/pull/4089#discussion_r1421831492) discussion. I strongly disagree with what you wrote there. I don't want to see API surface grow. It can always grow if there is a need, but it cannot ever shrink. We should start with small API and consider extending it. What I would rather do is create a ticket to discuss this and use `<ISI>::new` everywhere for now

you said:
> 1. Doesn't require expicit generics

 why is requiring explicit generics bad? But nevertheless, explicit generics are only required if the type you are providing to a function is not known. You don't have to use turbofish if the type of arguments is known

Reply:
>  But nevertheless, explicit generics are only required if the type you are providing to a function is not known. You don't have to use turbofish if the type of arguments is known

Yes, but as I said, in most of the time `<ISI>::new()` accepts not just a generic, but an associated type of this generic. That requires to **always** use turbo-fish. To me turbo-fish is less readable than a properly named method.

> I don't want to see API surface grow. It can always grow if there is a need, but it cannot ever shrink. We should start with small API and consider extending it.

I don't understand, where did it grow? All possible instructions are still the same, I haven't expand them.

I think the API is more clear with construction methods because user can open documentation for any instruction and easily see all possible *variants*. They don't need to understand what the heck is `<T as Registered>::With`. That can be a pain especially then for `Account` it's `NewAccount`, but for `Asset` it's just `Asset`.

> What I would rather do is create a ticket to discuss this and use <ISI>::new everywhere for now

Do you still consider this is an option keeping in mind that **all** i.e. transfers will look like:

```rust
let transfer = Transfer::<Asset, u32, Account>::new(asset_id, 1_u32, account_id);
```

Just... Should it be that hard?

Reply:
> Yes, but as I said, in most of the time <ISI>::new() accepts not just a generic, but an associated type of this generic. That requires to always use turbo-fish. To me turbo-fish is less readable than a properly named method.

I see. Let's have a vote. We have 2 options:
1.
```rs
let transfer = Transfer::asset_quantity(asset_id, 1_u32, account_id);
```

vs

2.
```rs
let transfer: Transfer<Asset, u32, Account> = Transfer::new(asset_id, 1_u32, account_id);
```

Reply:
Maybe we can remove associated types? 
In such case i would definitely vote for 2.

Reply:
let's leave it as it is